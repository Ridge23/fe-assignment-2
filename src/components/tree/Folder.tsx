import React, { useState } from 'react'
import classNames from 'classnames';

import DeleteIcon from 'components/icons/DeleteIcon';

import './sass/folder.scss'

interface IFolder {
    name: string;
    childrenNodes?: JSX.Element[];
    showDelete?: boolean;
    onDelete: () => void;
}

export default function Folder({ name, childrenNodes, showDelete = true, onDelete }: IFolder): JSX.Element {
    const [showFolder, setShowFolder] = useState(false);
    const noChildrenNodes = !childrenNodes || childrenNodes.length === 0
    const aClass = classNames([
        { 'folder__link--active': !noChildrenNodes },
        { 'folder__link--empty': noChildrenNodes },
    ])
    const liClass = classNames('folder__li', {'folder__li--active': showFolder});
    
    return (
        <>
            <li key={`folder-${name}`} className={liClass}>
                <a className={aClass} onClick={() => !noChildrenNodes && setShowFolder(!showFolder)}>{name}</a>
                <div className="folder__actions">
                    {showDelete && <DeleteIcon onClick={onDelete} />}
                </div>
            </li>
            {childrenNodes && showFolder &&
                <ul className="folder" key={`folder-children-${name}`}>
                    {childrenNodes.map((element) => element)}
                </ul>
            }
        </>
    );
}