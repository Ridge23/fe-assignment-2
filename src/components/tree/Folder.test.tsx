import React from 'react';
import { shallow } from 'enzyme';

import Folder from './Folder';

const noop = () => {};

describe('Folder', () => {
    it('should render correctly without children', () => {
        const component = shallow(<Folder 
            id={10} 
            name='test' 
            onDelete={noop} 
            onRename={noop} 
            onFileCreate={noop}
            onFolderCreate={noop}
        />);
        const li = component.find('li');

        expect(li).toHaveLength(1);
        expect(li.props()).toHaveProperty('className', 'folder__li');

        const a = component.find('a');
        expect(a).toHaveLength(1);
        expect(a.props()).toHaveProperty('className', 'folder__link--empty');
        expect(component).toMatchSnapshot();
    });
})