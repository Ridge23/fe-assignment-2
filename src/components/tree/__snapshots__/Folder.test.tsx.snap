// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder should render correctly without children 1`] = `
<Fragment>
  <li
    className="folder__li"
    key="folder-test"
  >
    <ContextMenuTrigger
      attributes={Object {}}
      collect={[Function]}
      disable={false}
      disableIfShiftIsPressed={false}
      holdToDisplay={1000}
      id="folder_menu_10"
      mouseButton={2}
      posX={0}
      posY={0}
      renderTag="div"
    >
      <a
        className="folder__link--empty"
        onClick={[Function]}
      >
        test
      </a>
    </ContextMenuTrigger>
    <FolderContextMenu
      id={10}
      onAddFileClick={[Function]}
      onAddFolderClick={[Function]}
      onDeleteClick={[Function]}
      onRenameClick={[Function]}
      showDelete={true}
    />
  </li>
</Fragment>
`;
